chore: switch to new shared tooling (`terraform()` pipeline & `shared-tools` common repo) (#8)
jenkins-infra/digitalocean: README.adoc

= Jenkins Infra on Digital Ocean
:toc:
:private_repo_name: terraform-states

> :private_repo_url: https://github.com/jenkins-infra/{private_repo_name}
> 

Reviewer: ```suggestion
:private_repo_url: https://github.com/jenkins-infra/{private_repo_name}
:shared_tools_repo_name: shared_tools
:shared_tools_repo_url: https://github.com/jenkins-infra/{shared_tools_repo_name}
```

This repository hosts the infrastructure-as-code definition for all the link:https://www.digitalocean.com/[Digital Ocean] resources for the link:https://www.jenkins.io/projects/infrastructure/[Jenkins Infrastructure Project].

== Requirements

* A Digital Ocean account with a personal access token defined as the value of the environment variable `TF_VAR_do_token`

> * The requirements (of the shared tools) listed at link:https://github.com/jenkins-infra/shared-tools/tree/main/terraform#requirements[shared-tools/terraform#requirements]

Reviewer: ```suggestion
* The requirements (of the shared tools) listed at link:{shared_tools_repo_url}/tree/main/terraform#requirements[{shared_tools_repo_name}/terraform#requirements]
```

* The link:https://www.terraform.io/docs/language/settings/backends/s3.html[Terraform S3 Backend Configuration] on a local file named `backend-config`:
** The content can be retrieved from the outputs of the link:{private_repo_url}[(private) repository {private_repo_name}]
** This file (`backend-config`) is git-ignored

> * The git command line to allow cloning the repository and its submodule link:https://github.com/jenkins-infra/shared-tools[shared-tools]

Reviewer: ```suggestion
* The git command line to allow cloning the repository and its submodule link:link:{shared_tools_repo_url}[{shared_tools_repo_name}]
```

> ** This repository has submodules. Once you cloned the repository, execute the following command to obtain the shared tools:

Reviewer: ```suggestion
** This repository has [submodules](https://git-scm.com/docs/git-submodule). Once you cloned the repository, execute the following command to retrieve the shared tools:
```

[source,bash]
----
git submodule update --init --recursive
----

== HowTo

=== Provision

IMPORTANT: Don't blindly execute the terraform code located in this repository on your own account as it may lead your account bill to significantly increase.

> Once you've fulfilled the <<Requirements>>, you may execute any command from https://github.com/jenkins-infra/shared-tools/blob/main/terraform/README.adoc#available-commands by adding the correct flag `--directory` pointing to `.shared-tools/terraform/`:

Reviewer: ```suggestion
Once you've fulfilled the <<Requirements>>, you may execute any command from {shared_tools_repo_url}/blob/main/terraform/README.adoc#available-commands by adding the correct flag `--directory` pointing to `.shared-tools/terraform/`:
```

[source,bash]
----
make --directory=.shared-tools/terraform help
make --directory=.shared-tools/terraform lint
# ...
----

> A usual change to this repository looks like the following:

Reviewer: ```suggestion
An usual workflow to update this repository looks like the following:
```

* Fork the repository and clone it locally

> * Follow the <<Requirements>> steps to obtain the shared tools

Reviewer: ```suggestion
* Follow the <<Requirements>> steps to retrieve the shared tools
```

* Start by running a full `make --directory=.shared-tools/terraform validate` command to ensure that you work on a sane base (should generate a report TXT file with no changes to be applied)
* Edit the Terraform project files

> * Run the command `make --directory=.shared-tools/terraform validate` again to ensure that your changes are OK

Reviewer: ```suggestion
* Run the command `make --directory=.shared-tools/terraform validate` again to review and ensure that your changes are OK
```

> * Commit, push and open a pull request to let the Jenkins pipeline run the test + plan (as per https://github.com/jenkins-infra/shared-tools/blob/main/terraform/README.adoc#jenkins-pipeline)

Reviewer: ```suggestion
* Commit, push and open a pull request to let the Jenkins pipeline run the test + plan (as per {shared_tools_repo_url}/blob/main/terraform/README.adoc#jenkins-pipeline)
```

= Jenkins Infra on Digital Ocean
:toc:
:private_repo_name: terraform-states
:private_repo_url: https://github.com/jenkins-infra/{private_repo_name}
:shared_tools_repo_name: shared_tools

> :shared_tools_repo_url: https://github.com/jenkins-infra/{shared_tools_repo_name}

Reviewer: ```suggestion
:shared_tools_repo_url: https://github.com/jenkins-infra/{shared_tools_repo_name}

```

= Jenkins Infra on Digital Ocean
:toc:
:private_repo_name: terraform-states
:private_repo_url: https://github.com/jenkins-infra/{private_repo_name}
:shared_tools_repo_name: shared_tools
:shared_tools_repo_url: https://github.com/jenkins-infra/{shared_tools_repo_name}

This repository hosts the infrastructure-as-code definition for all the link:https://www.digitalocean.com/[Digital Ocean] resources for the link:https://www.jenkins.io/projects/infrastructure/[Jenkins Infrastructure Project].

== Requirements

* A Digital Ocean account with a personal access token defined as the value of the environment variable `TF_VAR_do_token`
* The requirements (of the shared tools) listed at link:{shared_tools_repo_url}/tree/main/terraform#requirements[{shared_tools_repo_name}/terraform#requirements]
* The link:https://www.terraform.io/docs/language/settings/backends/s3.html[Terraform S3 Backend Configuration] on a local file named `backend-config`:
** The content can be retrieved from the outputs of the link:{private_repo_url}[(private) repository {private_repo_name}]
** This file (`backend-config`) is git-ignored

> * The git command line to allow cloning the repository and its submodule link:link:{shared_tools_repo_url}[{shared_tools_repo_name}]

Reviewer: ```suggestion
* The git command line to allow cloning the repository and its submodule link:{shared_tools_repo_url}[{shared_tools_repo_name}]
```
